Updated the OpenStack components install method

costingalan opened the pull request:
This commit adds the requirements check for the components, thus making
sure the requirements are synced by the OpenStack Requirements project
team.

The components affected by this change are:
- neutron
- networking-hyperv

ociuhandu commented on HyperV/scripts/create-environment.ps1:
```diff
@@ -137,7 +137,12 @@ if ($buildFor -eq "openstack/nova"){
         GitClonePull "$buildDir\networking-hyperv" "https://git.openstack.org/openstack/networking-hyperv.git" $branchName
     }
-}else{
+    if ($branchName -eq 'master') {
```
Why on master branch only?

ociuhandu commented on HyperV/scripts/create-environment.ps1:
```diff
@@ -222,13 +227,29 @@ if ($isDebug -eq  'yes') {
     Get-ChildItem $buildDir
 }
 
+if ($branchName -eq 'master') {
```
Why install only on master branch?

ociuhandu commented on HyperV/scripts/create-environment.ps1:
```diff
@@ -222,13 +227,29 @@ if ($isDebug -eq  'yes') {
     }
     pushd $buildDir\neutron
-    & pip install $buildDir\neutron
+    if ($branchName -eq 'master') {
```
ditto

ociuhandu commented on HyperV/scripts/create-environment.ps1:
```diff
@@ -239,7 +260,12 @@ ExecRetry {
     }
     pushd $buildDir\networking-hyperv
-    & pip install $buildDir\networking-hyperv
+    if ($branchName -eq 'master') {
```
ditto

ociuhandu commented on HyperV/scripts/create-environment.ps1:
```diff
@@ -255,7 +281,12 @@ ExecRetry {
     cherry_pick FETCH_HEAD
 
-    & pip install $buildDir\nova
+    if ($branchName -eq 'master') {
```
ditto

Commit:
Updated the OpenStack components install method

This commit adds the requirements check for the components, thus making
sure the requirements are synced by the OpenStack Requirements project
team. Also it will clear the manual pip installs

The components affected by this change are:

* neutron
* networking-hyperv
* os-win
* nova

Signed-off-by: Costin Galan <cgalan@cloudbasesolutions.com>

ociuhandu commented:
updated these requirements through other PRs
